tests/unit_node/http2_test.ts

```diff
@@ -164,3 +167,39 @@ Deno.test("[node/http2.createServer()]", {
   // Issue: https://github.com/denoland/deno/issues/22764
   await new Promise<void>((resolve) => server.on("close", resolve));
 });
+
+Deno.test("[node/http2 client] write image buffer on request stream works", async () => {
+  const url = "https://localhost:5545";
+  const client = http2.connect(url);
+  client.on("error", (err) => console.error(err));
+
+  const imagePath = join(import.meta.dirname!, "testdata", "green.jpg");
+  const buffer = await readFile(imagePath);
+  const req = client.request({ ":method": "POST", ":path": "/echo_server" });
+  req.write(buffer, (err) => {
+    if (err) throw err;
+  });
+
+  let receivedData: Buffer;
+  req.on("data", (chunk) => {
+    if (!receivedData) {
+      receivedData = chunk;
+    } else {
+      receivedData = Buffer.concat([receivedData, chunk]);
+    }
+  });
+  req.end();
+
+  const endPromise = Promise.withResolvers<void>();
+  setTimeout(() => {
+    try {
+      client.close();
+    } catch (_) {
+      // pass
+    }
+    endPromise.resolve();
+  }, 2000);
```

Comment:
The test will fail if recieving takes more than 2s. Maybe compare recievedData.length == buffer.length in `req.on('data')` and resolve the endPromise.

Reply:
The timeout acts a circuit breaker - if the the response is not received within the specified time then something is wrong and we want to break instead of waiting for the response to avoid hanging tests.

For this specific test, the response should be available in the first chunk itself.